pythran/pythonic/types/tuple.hpp

```diff
@@ -296,11 +311,11 @@ namespace pythonic {
 
                 template<class... Types>
                     operator std::tuple<Types...>() const {
-                        return std::tuple<Types...>(to_tuple_impl<N>()(*this));
+                        return array_to_tuple(*this, typename utils::gens<N+1>::type{});
                     }
 
-                auto to_tuple() const -> decltype(to_tuple_impl<N>()(*this)) {
-                    return to_tuple_impl<N>()(*this);
```

Comment:
I like the new implementation but to_tuple_impl should be remove in this case. (It is not use anymore)